transforms/code/proglang_select/kfp_ray/proglang_select_wf.py

```diff
@@ -10,32 +10,75 @@
 # limitations under the License.
 ################################################################################
 
+import os
+
+from workflow_support.compile_utils import ONE_HOUR_SEC, ONE_WEEK_SEC, ComponentUtils
+
 import kfp.compiler as compiler
 import kfp.components as comp
 import kfp.dsl as dsl
-from kfp_support.workflow_support.utils import (
-    ONE_HOUR_SEC,
-    ONE_WEEK_SEC,
-    ComponentUtils,
-)
 
 
 # the name of the job script
 EXEC_SCRIPT_NAME: str = "proglang_select_transform_ray.py"
 
-task_image = "quay.io/dataprep1/data-prep-kit/proglang_select-ray:0.4.0"
+task_image = "quay.io/dataprep1/data-prep-kit/proglang_select-ray:0.4.0.dev6"
 
 # components
-base_kfp_image = "quay.io/dataprep1/data-prep-kit/kfp-data-processing:0.2.0"
+base_kfp_image = "quay.io/dataprep1/data-prep-kit/kfp-data-processing:0.2.0.dev6"
 
 # path to kfp component specifications files
-component_spec_path = "../../../../../kfp/kfp_ray_components/"
+component_spec_path = "../../../../kfp/kfp_ray_components/"
+
 
 # compute execution parameters. Here different tranforms might need different implementations. As
 # a result, instead of creating a component we are creating it in place here.
-compute_exec_params_op = comp.func_to_container_op(
-    func=ComponentUtils.default_compute_execution_params, base_image=base_kfp_image
-)
+def compute_exec_params_func(
+    worker_options: str,
+    actor_options: str,
+    data_s3_config: str,
+    data_max_files: int,
+    data_num_samples: int,
+    runtime_pipeline_id: str,
+    runtime_job_id: str,
+    runtime_code_location: str,
+    proglang_select_allowed_langs_file: str,
+    proglang_select_language_column: str,
+) -> dict:
+    from workflow_support.runtime_utils import KFPUtils
+
+    return {
+        "data_s3_config": data_s3_config,
+        "data_max_files": data_max_files,
+        "data_num_samples": data_num_samples,
+        "runtime_num_workers": KFPUtils.default_compute_execution_params(worker_options, actor_options),
+        "runtime_worker_options": actor_options,
+        "runtime_pipeline_id": runtime_pipeline_id,
+        "runtime_job_id": runtime_job_id,
+        "runtime_code_location": runtime_code_location,
+        "proglang_select_allowed_langs_file": proglang_select_allowed_langs_file,
+        "proglang_select_language_column": proglang_select_language_column,
+    }
+            
```

Comment:
Can we move this to the library?

Reply:
We cannot put all dictionary entries into a lib, because each pipeline has different parameters. 
We can put the common part, but it can complicate readability.